### PR TITLE
Lab6

### PR DESCRIPTION
**Pull request by Reqvel**
Having a hard time deploying this project, so i included demo images

### CODE REVIEW

File: lab6/src/App.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          <Route path="/MusicPlayer" element={ <PrivateRoute /> }>
+            <Route path="/MusicPlayer" element={ <PlayerPage /> }>
+              <Route path="Playlists" element={ <PlaylistsPage /> }>
+                <Route path="" element={ <Playlists /> } />
+                <Route path="Tracks" element={ <PlaylistTracks /> } />
+              </Route>
+              <Route path="Search" element={ <SearchPage /> } />
+            </Route>
+          </Route>
+        </Routes>
```
Comment (katetushkan) [inline lab6/src/App.js]: Place it in a detached file.

File: lab6/src/common/GetArtistsNames.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export function getArtistsNames(artists) {
+    var names = ''
```
Comment (katetushkan) [inline lab6/src/common/GetArtistsNames.js]: let.

File: lab6/src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { Link } from "react-router-dom"
+import { useState } from 'react';
+
+const Header = () => {
+  const [isShown, setIsShown] = useState(false)
+  const showHideSideNav = () => setIsShown(!isShown)
+
+  return (
+    <div className="header-container">
+      <div className="header">
```
Comment (katetushkan) [inline lab6/src/components/Header.jsx]: maybe, if it's `header` than it should be `<header>` ?)

File: lab6/src/components/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                <li>
+                    <a href="https://www.spotify.com/us/signup" className="a text nav-text">Sign Up</a>
+                </li>
+            </ul>
+        </nav>
+        <button 
+          className={isShown ? "menu-button menu-button-media cursor-pointer menu-button-selected" : "menu-button menu-button-media cursor-pointer"}
+          onClick={showHideSideNav}>
+          <MenuIconS />
+        </button>
```
Comment (katetushkan) [inline lab6/src/components/Header.jsx]: Should be a separate component. with class `button`, and props, which is gonna decide how the classname gonna looks like 

```
interface IProps {
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
  autoFocus?: boolean;
  value?: string;
  disabled?: boolean;
  type?: 'normal' | 'accent';
  to?: string;
}

class Button extends React.Component<IProps> {

  render() {
    const { onClick, className, children, autoFocus, value, disabled, type, to } = this.props;

    return (
      <>
        {to
        ? <Link
            to={to}
            className={clsx('button', {
              'button--normal': !type || type === 'normal',
              'button--accent': type === 'accent',
              'button--disabled': disabled,
            }, className)}
            onClick={onClick}
          >
            {children}
          </Link>
        : <button
            autoFocus={autoFocus}
            className={clsx('button', {
              'button--normal': !type || type === 'normal',
              'button--accent': type === 'accent',
              'button--disabled': disabled,
            }, className)}
            onClick={onClick}
            value={value}
            disabled={disabled}
          >
            {children}
          </button>
        }
      </>
    )
  }
}

export default Button;
```

Smth like this.

File: lab6/src/components/MusicPlayerControls/NextBtn.jsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { ReactComponent as Next } from '../../svgs/Next.svg'
+
+const NextBtn = ({ onClick }) => {
+    return (
+        <button className="controls-button controls-button-next" onClick={onClick}>
+            <Next />
+        </button>
+    )
+}
+
```
Comment (katetushkan) [inline lab6/src/components/MusicPlayerControls/NextBtn.jsx:10]: All ur buttons should be a '<Button/>' but u can manipulate its appearance w/ outer styles and children props, and behaviour with onClick callback

**Review by katetushkan — CHANGES_REQUESTED** [review]
Pls, try to deploy cuz I can't check it properly without it, maybe with GitHub or firebase it's the most basic ways.

**Comment by Reqvel** [comment]
alright i will try to deploy it, but i need ur email that's attached to ur Spotify account(i need it to add to devs) and Spotify Premium is required for full access

**Comment by Reqvel** [comment]
deploy https://spotifee-99d72.web.app/

**Comment by katetushkan** [comment]
> alright i will try to deploy it, but i need ur email that's attached to ur Spotify account(i need it to add to devs) and Spotify Premium is required for full access

So I can't login with my account by myself ?

**Comment by Reqvel** [comment]
yes, u can, but the API will not respond to anything because your account is not in the list of developers. In order to properly deploy the app, I need the approval of the Spotify app review team. And they will not approve it :)

**Pull request closed without merging**